store/rootmulti/store.go

```diff
@@ -672,6 +712,10 @@ func (rs *Store) GetKVStore(key types.StoreKey) types.KVStore {
 	return store
 }
 
+func (rs *Store) GetMemStore() types.MemStore {
+	return rs.memStoreManager.Branch()
```

Comment:
why do we branch here, but not in cachemulti? Why do we even branch when we just retrieve Memstore?